dagshub/data_engine/client/data_client.py

```diff
@@ -183,6 +244,16 @@ def save_dataset(self, datasource: Datasource, name: str):
         return self._exec(q, params)
 
     def get_datasets(self, id: Optional[Union[str, int]], name: Optional[str]) -> List[DatasetResult]:
+        """
+        Retrieve a list of datasets based on optional filtering criteria.
```

Comment:
@kbolashev doe we have a unified function to search both datasources and datasets?

I think we should, since the distinction might be confusing to users.

Reply:
Impossible because there could be name collisions. You can name a dataset same as the datasource, and getting them by name would lead to unexpected behaviors